modules/tinymce/src/core/test/ts/browser/keyboard/ArrowKeysInlineBoundariesTest.ts

```diff
@@ -336,4 +336,27 @@ describe('browser.tinymce.core.keyboard.ArrowKeysInlineBoundariesTest', () => {
       TinyAssertions.assertCursor(editor, [ 0, 0 ], 0);
     });
   });
+
+  context('Arrow keys on inline boundary node when selection is not collapsed', () => {
+    it('TINY-8601: It should move caret before selected inline boundary node when arrow left', () => {
+      const editor = hook.editor();
+      editor.setContent('<p>test <span class="mce-annotation">span</span> selection</p>', { format: 'raw' });
+      TinySelections.select(editor, 'p', [ 1 ]);
```

Comment:
If you use the `setSelection` api then it has a extra parameter to fire a node change. So maybe change this to:
`TinySelections.setSelection(editor, [0], 1, [0], 2, true);`
But then again the `TinySelections.select` call here is probably easier to understand than the paths above not sure. I just tend to like using the `setSelection` api my self since it's more exact of what it's doing you kind of see both end points where they are going.